src/test/java/com/team4/testingsystem/services/TestsServiceTest.java

```diff
@@ -0,0 +1,189 @@
+package com.team4.testingsystem.services;
+
+import com.team4.testingsystem.entities.Test;
+import com.team4.testingsystem.entities.User;
+import com.team4.testingsystem.exceptions.TestNotFoundException;
+import com.team4.testingsystem.exceptions.UserNotFoundException;
+import com.team4.testingsystem.repositories.TestsRepository;
+import com.team4.testingsystem.repositories.UsersRepository;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.dao.EmptyResultDataAccessException;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class TestsServiceTest {
+
+    @Mock
+    Test test;
+
+    @Mock
+    Test.Builder builder;
+
+    @Mock
+    User user;
+
+    @Mock
+    UsersRepository usersRepository;
+
+    @Mock
+    TestsRepository testsRepository;
+
+    @InjectMocks
+    TestsService testsService;
+
+
+    @org.junit.jupiter.api.Test
+    void getAllSuccess() {
+        List<Test> tests = new ArrayList<>();
+        Mockito.when(testsRepository.findAll()).thenReturn(tests);
+
+        Assertions.assertEquals(tests, testsService.getAll());
+
+    }
+
+    @org.junit.jupiter.api.Test
+    void getByIdSuccess() {
+
+        Mockito.when(testsRepository.findById(1L)).thenReturn(Optional.of(test));
+
+        Assertions.assertEquals(test, testsService.getById(1L));
+
+    }
+
+    @org.junit.jupiter.api.Test
+    void getByIdFail() {
+
+        //Test doesn't exist
+        Mockito.when(testsRepository.findById(42L)).thenThrow(NoSuchElementException.class);
+
+        Assertions.assertThrows(TestNotFoundException.class, () -> testsService.getById(42L));
+
+    }
+
+    @org.junit.jupiter.api.Test
+    void createWhenAssignFail() {
+
+        //User doesn't exist
+        Mockito.when(usersRepository.findById(42L)).thenThrow(NoSuchElementException.class);
+
+        Assertions.assertThrows(UserNotFoundException.class, () -> testsService.createForUser(42L));
+    }
+
+    @org.junit.jupiter.api.Test
+    void createSuccess() {
+
+
+        Mockito.when(usersRepository.findById(1L)).thenReturn(Optional.of(user));
+
+        MockedStatic<Test> testMockedStatic = Mockito.mockStatic(Test.class);
+        testMockedStatic.when(Test::newBuilder).thenReturn(builder);
```

Comment:
Mockito in it's documentation recommends to define the scope within a try-with-resources construct for [mocking static methods](https://javadoc.io/static/org.mockito/mockito-core/3.11.2/org/mockito/Mockito.html#static_mocks).
So, static mocks can be used like this:
```
try (MockedStatic<Test> mockTest = Mockito.mockStatic(Test.class)) {
    <use mockTest in this scope>
}
```